pageserver/src/tenant.rs

```diff
@@ -1371,9 +1406,39 @@ impl Tenant {
         timeline.walreceiver.stop().await;
         debug!("wal receiver shutdown confirmed");
 
+        // Prevent new uploads from starting.
+        if let Some(remote_client) = timeline.remote_client.as_ref() {
+            let res = remote_client.stop();
+            match &res {
+                Ok(()) => {}
+                Err(e) => match e {
+                    remote_timeline_client::StopError::QueueUninitialized => {
+                        // This could happen if the timeline is Broken, e.g., because it failed to fetch IndexPart when it was loaded.
+                    }
+                },
+            }
+            res?;
+        }
+
+        // Stop & wait for the remaining timeline tasks, including upload tasks.
```

Comment:
If there is already a delete in progress for this timeline, does this kill the task that's performing the deletion?

Reply:
No, because callers are assumed to be http request handlers.

Reply:
Added a clarifying comment.